### PR TITLE
[3006.x] Add salt.ufw to salt-master install on Debian and Ubuntu

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Adds salt.ufw back to the salt-master install for Debian and Ubuntu

With the transition to using GitHub Actions, salt.ufw was removed fro, salt-master.install incorrectly
Also removed upstart.rhel6 which should have never removed, as RHEL 6 is long EOL

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/64572

### Previous Behavior
sudo ufw allow salt should would run with errors

### New Behavior
sudo ufw allow salt should run without errors

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by s0undt3ch — APPROVED** [review]

**Review by cmcmarrow — APPROVED** [review]

File: tests/pytests/integration/daemons/test_masterapi.py [hunk 1]
```diff
@@ -120,3 +121,23 @@ def test_autosign_grains_fail(
...
+def test_ufw_allow(salt_master, grains):
+    if grains["os_family"] != "Debian":
+        pytest.skip("Only runs on Debian family.")
+
+    expected_output = """Skipping adding existing rule
+Skipping adding existing rule (v6)
+    """
+
+    proc = subprocess.Popen(
+        "ufw allow salt\n", shell=True, stdin=subprocess.PIPE, stdout=subprocess.PIPE
```
Comment (Ch3LL) [inline tests/pytests/integration/daemons/test_masterapi.py:137]: Question: If this is a package change, wouldn't the test need to be in the package test suite?

Comment (dmurphy18) [reply tests/pytests/integration/daemons/test_masterapi.py:137]: You raise an interesting point.  I wanted to be able to test this without things having been packaged, but I should probably add a test there too, or move the test there.

Will look into it.

**Comment by rousku** [comment]
This doesn't work. salt.ufw should be installed to /etc/ufw/applications.d.
```
ubuntu@ubuntu:~/git/ssc600-e4s-salt$ ls /etc/ufw/applications.d/salt-master/
salt.ufw
ubuntu@ubuntu:~/git/ssc600-e4s-salt$ sudo ufw app list
Available applications:
  CUPS
  OpenSSH
ubuntu@ubuntu:~/git/ssc600-e4s-salt$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.3 LTS
Release:	22.04
Codename:	jammy
ubuntu@ubuntu:~/git/ssc600-e4s-salt$ dpkg -l | grep salt-master
ii  salt-master                                                 3006.3                                     amd64        remote manager to administer servers via salt
ubuntu@ubuntu:~/git/ssc600-e4s-salt$ dpkg-query -L salt-master | grep ufw
/etc/ufw
/etc/ufw/applications.d
/etc/ufw/applications.d/salt-master
/etc/ufw/applications.d/salt-master/salt.ufw
ubuntu@ubuntu:~/git/ssc600-e4s-salt$ 
```

**Comment by dmurphy18** [comment]
@rousku see https://github.com/saltstack/salt/issues/64572#issuecomment-1726025386

**Pull request merged**